src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,48 @@
 import React from 'react';
 import './PostsList.scss';
+import { Post } from '../../types/post';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+type Props = {
+  posts: Post[],
+  selectedPostId: number,
+  handleSelectedPostId: (postId: number) => void,
+};
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+export const PostsList: React.FC<Props> = React.memo(({
+  posts,
+  selectedPostId,
+  handleSelectedPostId,
+}) => {
+  const onPostIdChange = (postId: number) => {
+    if (selectedPostId !== postId) {
+      handleSelectedPostId(postId);
+    } else {
+      handleSelectedPostId(0);
+    }
```

Comment:
why do you even need this `else` block?
```suggestion
    }
```

Reply:
To download all posts of all users from the server. But if the option `all users` is disabled, then it makes no sense